sdk/core/azure-core/test/ut/curl_session_test.cpp

```diff
@@ -82,6 +82,130 @@ namespace Azure { namespace Core { namespace Test {
     Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.clear();
   }
 
+  TEST_F(CurlSession, chunkBadFormatResponse)
+  {
+    // chunked response with unexpected char at the end
+    std::string response("HTTP/1.1 200 Ok\r\ntransfer-encoding: chunked\r\n\r\n9\r\n");
+    std::string response2("123456789\r\n0\r\n\rx");
+    std::string connectionKey("connection-key");
+
+    // Can't mock the curMock directly from a unique ptr, heap allocate it first and then make a
+    // unique ptr for it
+    MockCurlNetworkConnection* curlMock = new MockCurlNetworkConnection();
+    EXPECT_CALL(*curlMock, SendBuffer(_, _, _)).WillOnce(Return(CURLE_OK));
+    EXPECT_CALL(*curlMock, ReadFromSocket(_, _, _))
+        .WillOnce(DoAll(
+            SetArrayArgument<1>(response.data(), response.data() + response.size()),
+            Return(response.size())))
+        .WillOnce(DoAll(
+            SetArrayArgument<1>(response2.data(), response2.data() + response2.size()),
+            Return(response2.size())));
+    EXPECT_CALL(*curlMock, GetConnectionKey()).WillRepeatedly(ReturnRef(connectionKey));
+    EXPECT_CALL(*curlMock, updateLastUsageTime());
+    EXPECT_CALL(*curlMock, DestructObj());
+
+    // Create the unique ptr to take care about memory free at the end
+    std::unique_ptr<MockCurlNetworkConnection> uniqueCurlMock(curlMock);
+
+    // Simulate a request to be sent
+    Azure::Core::Http::Url url("http://microsoft.com");
+    Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Get, url);
+
+    {
+      // Create the session inside scope so it is released and the connection is moved to the pool
+      auto session = std::make_unique<Azure::Core::Http::CurlSession>(
+          request, std::move(uniqueCurlMock), true);
+
+      EXPECT_NO_THROW(session->Perform(Azure::Core::GetApplicationContext()));
+      auto response = session->GetResponse();
```

Comment:
nit: We end up using the same variable name as above `std::string response` outside the nested scope, consider renaming one.

Reply:
that's fine, as long as it is in a inner scope it's fine